lib/system/base/network.toit

```diff
@@ -45,16 +45,67 @@ class NetworkResource extends ServiceResourceProxy:
     socket ::= client.tcp_listen handle_ port
     return TcpServerSocketResourceProxy_ client socket
 
+// ----------------------------------------------------------------------------
+
+interface NetworkModule:
+  connect -> none
+  disconnect -> none
+
+class NetworkResource extends ServiceResource:
+  state_/NetworkState ::= ?
+  constructor service/ServiceDefinition client/int .state_ --notifiable/bool=false:
+    super service client --notifiable=notifiable
+  on_closed -> none:
+    critical_do: state_.down
+
+monitor NetworkState:
```

Comment:
This could use a line or two of comment to indicate what it is for.

Reply:
Done.